core/src/main/java/hudson/util/io/TarArchiver.java

```diff
@@ -59,7 +59,7 @@
     public void visitSymlink(File link, String target, String relativePath) throws IOException {
         TarArchiveEntry e = new TarArchiveEntry(relativePath, TarConstants.LF_SYMLINK);
         try {
-            int mode = IOUtils.mode(link);
```

Comment:
> Well, you were quotinq that one single line change

I wasn't quoting anything. I was responding to your comment about Jenkins' current behavior and how it should behave differently. GitHub has a feature that allows multiple conversations about the same line (it's the button that says *Start a new conversation*), and I did not start a new one, but deliberately responded to your comment.

> After that - you have the whole jenkins dir in your archive - including the "secrets"

This looks bad until you consider that anyone able to do this can extract the secrets any way they please, so preventing one of a million different ways to extract them is a waste of time. Which is why one of the most important security advice for admins is to not build on the Jenkins master (with the possible exception of things like backup jobs that must run on master, and don't process any user input). See e.g. https://wiki.jenkins-ci.org/display/JENKINS/Jenkins+Best+Practices

> add a checkbox "Yes, I want the old, broken behavior" somewhere in the GUI

Probably not an unreasonable approach (if worded less snidely). It could also be made the default for new job configurations. Probably best discussed further in the context of a PR that implements this.